doc/reference/std/errors.md

```diff
@@ -1,64 +1,421 @@
+  err := instance of Error
 ```
-(defstruct (io-error <error>) ())
+
+Accessor for the error message.
+
+### Error-irritants
+```scheme
+(Error-irritants err) -> list
+  err := instance of Error
 ```
 
-Errors raised by the standard library because of I/O conditions.
+Accessor for the error irritants.
 
-## raise-io-error
+### Error-where
+```scheme
+(Error-where err) -> any
+  err := instance of Error
+```
 
-::: tip usage
+Accessor for the error origin.
+
+### Error:::init!
+```scheme
+(Error:::init! err message irritants: (irritants []) where: (where #f))
+  err := instance of Error
 ```
-(raise-io-error where what . irritants)
+
+The base `Error` constructor, which error subclasses can call to
+initialize the base error reprsentation.
+
+### Error::display-exception
+```scheme
+(Error::display-exception err port)
+  err  := instance of Error
+  port := output port
 ```
-:::
 
-Raises an `io-error`.
+Base method for displaying Errors, including a stack trace if
+available; see `StackTrace` below.
+
+
+## Standard Error Classes
 
-## timeout-error
+The following are predefined classes with standard semantics, used
+througout the standard library.
 
+### StackTrace
+```scheme
+(defclass StackTrace (continuation))
 ```
-(defstruct (timeout-error <error>) ())
+
+Mixin class for stack traces in errors.
+
+If an error class mixes in `StackTrace` is raised by this module's
+redefined `raise` (see below), then the current continuation will be
+captured (if not already captured) and stored in the `continuation`
+slot. The continuation backtrace will be displayed by
+`Error::display-exception` to facilitate debugging.
+
+
+### StackTrace?
+```scheme
+StackTrace?
 ```
 
-Errors raised by the standard library because of timeouts.
+Predicate for `StackTrace` mixins.
 
-## raise-timeout
+### BadArgument
+```scheme
+(defclass (BadArgument StackTrace Error) ())
+```
 
-::: tip usage
+Error indicating a bad argument condition.
+
+### BadArgument?
+```scheme
+(BadArgument? exn)
+```
+
+Predicate testing whether the error is a bad argument condition.
+
+### bad-argument-error?
+```scheme
+(def bad-argument-error? BadArgument?
+```
+
+same as `BadArgument?`
+
+### IOError
+```scheme
+(defclass (IOError StackTrace Error) ())
+```
+
+Error indicating an IO condition.
+
+### IOError?
+```scheme
+(IOError? exn)
+```
+
+Predicate testing whether the error is an IO condition.
+
+### io-error?
+```scheme
+(def io-error? IOError?
+```
+
+same as `IOError?`
+
+### PrematureEndOfInput
+```scheme
+(defclass (PrematureEndOfInput IOError) ())
+```
+
+Error indicating that input was exhausted prematurely; similar to EOF
+exceptions in other languages.
+
+### PrematureEndOfInput?
+```scheme
+(PrematureEndOfInput? obj)
+```
+
+Predicate testing whether the error is a premature end of input condition.
+
+### premature-end-of-input-error?
+```scheme
+(def permature-end-of-input? PrematureEndOfInput?)
+```
+
+Same as `PrematureEndOfInput?`.
+
+### IOClosed
```

Comment:
Why is it IOClosed and not Closed or CloserClosed or something not specific to IO sinks and sources? Anything Closer can be closed.

Reply:
Ok, Closed it is, i will rename.